src/compiler/utilities.ts

```diff
@@ -5075,11 +5075,18 @@ namespace ts {
         readonly isImplements: boolean;
     }
     export function tryGetClassImplementingOrExtendingExpressionWithTypeArguments(node: Node): ClassImplementingOrExtendingExpressionWithTypeArguments | undefined {
-        return isExpressionWithTypeArguments(node)
-            && isHeritageClause(node.parent)
-            && isClassLike(node.parent.parent)
-            ? { class: node.parent.parent, isImplements: node.parent.token === SyntaxKind.ImplementsKeyword }
-            : undefined;
+        if (isExpressionWithTypeArguments(node)) {
+            if (isHeritageClause(node.parent) && isClassLike(node.parent.parent)) {
+                return { class: node.parent.parent, isImplements: node.parent.token === SyntaxKind.ImplementsKeyword };
+            }
+            if (isJSDocAugmentsTag(node.parent)) {
+                const host = getEffectiveJSDocHost(node.parent);
+                if (host && isClassDeclaration(host)) {

```

Comment:
should this work with class expressions too? can you add a test and maybe switch this to `isClassLike(host)` if it doesn't work?